modules/nf-commons/src/main/nextflow/util/StringUtils.groovy

```diff
@@ -47,6 +50,42 @@ class StringUtils {
         return m.matches() ? m.group(1).toLowerCase() : null
     }
 
+    static private Pattern multilinePattern
+    static private List<String> maskPatterns = new ArrayList<>()
+    static private void addMaskPattern(String maskPattern) {
+        maskPatterns.add(maskPattern)
+        multilinePattern = Pattern.compile(maskPatterns.join("|"), Pattern.MULTILINE)
+    }
+
+    static{
+        addMaskPattern("password\\s*:\\s*\"(.*?)\"")
+        addMaskPattern("\"password\"\\s*:\\s*\"(.*?)\"")
+        addMaskPattern("token\\s*:\\s*\"(.*?)\"")
+        addMaskPattern("\"token\"\\s*:\\s*\"(.*?)\"")
+        addMaskPattern("secret\\s*:\\s*\"(.*?)\"")
+        addMaskPattern("\"secret\"\\s*:\\s*\"(.*?)\"")
+        addMaskPattern("licence\\s*:\\s*\"(.*?)\"")
+        addMaskPattern("\"licence\"\\s*:\\s*\"(.*?)\"")
+    }
+
+    static String stripSecrets(String message) {
+        if (multilinePattern == null) {
+            return message;
```

Comment:
Not understanding the need for this

Reply:
do you say the `if`? 
well, it's only a safeguard in the case for whatever reason someday we remove the static block who initializes the list

can be removed of course

Reply:
yes, but then why it's set [here](https://github.com/nextflow-io/nextflow/pull/2908/files#diff-3e31ee682cd3154c236f35dd401e69d8094a313d9880ef1ae55e94304e57530dR57)? maybe it can be just removed?

Reply:
(I had the feeling I wrote a similar comment, but can't see it, anyhow) Still understanding the logic of using this static pattern that's the last created by `addMaskPattern` ?

https://github.com/nextflow-io/nextflow/pull/2908/files#diff-3e31ee682cd3154c236f35dd401e69d8094a313d9880ef1ae55e94304e57530dR76

Reply:
basically, the idea is that the static block is creating a *multiline* regular expression as `"password": value` or `password  : "value"`  with every `addMaskPattern` 

after when we want to stripe a string we iterate over the multilinePattern  groups replacing occurrences with `*` 

Other similar methods we have are oriented to work with Map but the intent of this is to work with raw Strings

(not sure if it's what you are asking)

Reply:
Looks a bit convoluted, would not much easier doing something like `secret|password|blahblah\s*:\s*"(.*?)"` 

Also not the format is `password = 'something'` or `password = "something"`  (not with `:`, it's a groovy config basically)

Reply:
true, I missed the equals format

To me is easier to read and maintain a list of expressions than a big expression, but no problem, I'll change it